cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -2236,10 +2369,166 @@ def do_setup(self, context):
+                if vol['status'] in ['in-use', 'available']:
+                    # do we need to update the host?
+                    pool_info = volume_utils.extract_host(
+                        vol.host, level='pool', default_pool_name=True)
+
+                    # IF the pool has already been set correctly, then
+                    # no need to make api calls to vcenter to fetch the
+                    # datastore name from the volume backing information.
+                    # this will save time on every startup
+                    if (pool_info not in ds_keys or
+                            pool_info == volume_utils.DEFAULT_POOL_NAME):
+                        LOG.debug("Updating host for volume {}".format(vol.id))
+
+                        try:
+                            new_host = self._new_host_for_volume(vol)
+                            if new_host:
+                                vol.update({'host': new_host})
+                                vol.save()
+                        except Exception as ex:
+                            LOG.warning("Couldn't update host for {} because "
+                                        " {}".format(vol.id, ex))
+                    else:
+                        LOG.debug("Keeping host for volume {}".format(vol.id))
+
+            LOG.info("HOST COMPLETE {}".format(self.host))
+            return vol_updates, None
+        else:
+            # Since pools are not enabled, we should ensure that the datastore
+            # isn't part of the host.  This allows us to go backwards to
+            # not using datastores as pools.
+            LOG.info("vmware_datastores_as_pools is disabled.")
+
+            vol_updates = []
+            for vol in volumes:
+                # make sure we have the correc host info
+                if vol['status'] in ['in-use', 'available']:
+                    # do we need to update the host?
+                    pool_info = volume_utils.extract_host(
+                        vol.host, level='pool')
+                    backend_info = volume_utils.extract_host(
+                        vol.host, level='backend'
+                    )
+                    backend = backend_info.split("@")
+                    LOG.info("Volume host '{}' backend '{}'  pool '{}'".format(
+                        vol.host, backend[1], pool_info))
+
+                    # we need to force the host back to
+                    # host@backend#backend
+                    new_host = "{}@{}#{}".format(
+                        backend[0],
+                        backend[1],
+                        backend[1]
+                    )
+                    if new_host != vol.host:
+                        LOG.info("Setting host to {}".format(new_host))
+                        vol.update({'host': new_host})
+                        vol.save()
+
+            LOG.info("HOST COMPLETE {}".format(self.host))
+            return vol_updates, None
```

Comment:
Same here, why do we have `vol_updates`, if we never add anything?

Reply:
Fixed.